swarm/src/lib.rs

```diff
@@ -841,23 +834,21 @@ where
         })
 }
 
-impl<TBehaviour, TInEvent, TOutEvent, THandler> Stream for
+impl<TBehaviour, TInEvent, TOutEvent, THandler, THandleErr> Stream for
     ExpandedSwarm<TBehaviour, TInEvent, TOutEvent, THandler>
 where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
       THandler: IntoProtocolsHandler + Send + 'static,
       TInEvent: Send + 'static,
       TOutEvent: Send + 'static,
-      THandler::Handler: ProtocolsHandler<InEvent = TInEvent, OutEvent = TOutEvent>,
+      THandler::Handler: 
+        ProtocolsHandler<InEvent = TInEvent, OutEvent = TOutEvent, Error = THandleErr>,
+      THandleErr: error::Error + Send + 'static,
 {
-    type Item = TBehaviour::OutEvent;
+    type Item = SwarmEvent<TBehaviour::OutEvent, THandleErr>;
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        loop {
-            let event = futures::ready!(ExpandedSwarm::poll_next_event(self.as_mut(), cx));
-            if let SwarmEvent::Behaviour(event) = event {
-                return Poll::Ready(Some(event));
-            }
-        }
+        let event = futures::ready!(ExpandedSwarm::poll_next_event(self.as_mut(), cx));
+        Poll::Ready(Some(event))
```

Comment:
```suggestion
        ExpandedSwarm::poll_next_event(self.as_mut(), cx).map(Some)
```

This should work as well. What do you think?

Or we could even go as far as the below, right?

```rust
self.as_mut()
    .poll_next_event(cx)
    .map(Some)
```